app/Http/Controllers/Import/Instagram.php

```diff
@@ -21,6 +22,13 @@ public function instagram()
 
     public function instagramStart(Request $request)
     {	
+        $completed = ImportJob::whereProfileId(Auth::user()->profile->id)
+            ->whereService('instagram')
+            ->whereNotNull('completed_at')
+            ->exists();
+        if($completed == true) {
+            return redirect(route('settings'))->with(['errors' => ['You can only import from Instagram once.']]);
```

Comment:
Is the plan here that once the user has completed their Instagram import they should use Pixelfed forever?

Reply:
hmm. yeah, idk about this, it should be possible to import/export multiple times ideally. that would require duplicate checking and partial archives, but it's much better than saying that you can only import once.

Reply:
@Haydos585 That is a good point, I will investigate a solution.

Reply:
You could try and limit it to once a month or something. I think with a service like Pixelfed if people are going to use it they will come, add some images, leave for 6 months, try it again, leave for 3 months, try it again, stick with it.

Having the same images imported multiple times is a problem too.

Perhaps something like `from_instagram` and `instagram_id` could be added to the Post table. Code can then figure out if something has been imported before.